lib/dal/src/func.rs

```diff
@@ -536,23 +535,50 @@ impl Func {
             )
             .await?;
 
-        let mut func_node_weights = vec![];
-        let mut func_content_hash = vec![];
+        let mut func_node_weights = Vec::new();
+        let mut func_content_hashes = Vec::new();
         for index in func_node_indexes {
             let node_weight = workspace_snapshot
                 .get_node_weight(index)
                 .await?
                 .get_func_node_weight()?;
-            func_content_hash.push(node_weight.content_hash());
+            func_content_hashes.push(node_weight.content_hash());
             func_node_weights.push(node_weight);
         }
 
+        Self::list_inner(ctx, func_node_weights, func_content_hashes).await
+    }
+
+    /// List all [`Funcs`](Func) corresponding to the provided [`FuncIds`](Func).
+    pub async fn list_from_ids(ctx: &DalContext, func_ids: &[FuncId]) -> FuncResult<Vec<Self>> {
```

Comment:
Break this part out for optimization so that other functions from other structs that collect `FuncIds` can use it.